sample_size/metrics.py

```diff
@@ -31,6 +34,10 @@ def check_positive(number: float, name: str) -> float:
         else:
             return number
 
+    @abstractmethod
+    def generate_p_value(self, true_alt: bool, size: int, variants: int) -> Any:
```

Comment:
it is not clear to me to directly know the meaning of each parameters for `true_alt`, `size`, and `variants`.  Maybe worth a docstring?
alt = alternative?

Reply:
+1. or we can make the variables names longer and clearer